instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -105,43 +114,40 @@ def _start_response(status, response_headers, *args, **kwargs):
     return _wrapped_app
 
 
-def _before_request():
-    if _excluded_urls.url_disabled(flask.request.url):
-        return
+def _wrapped_before_request(name_callback):
+    def _before_request():
+        if _excluded_urls.url_disabled(flask.request.url):
+            return
 
-    environ = flask.request.environ
-    span_name = None
-    try:
-        span_name = flask.request.url_rule.rule
-    except AttributeError:
-        pass
-    if span_name is None:
-        span_name = otel_wsgi.get_default_span_name(environ)
-    token = context.attach(
-        propagators.extract(otel_wsgi.carrier_getter, environ)
-    )
-
-    tracer = trace.get_tracer(__name__, __version__)
-
-    span = tracer.start_span(
-        span_name,
-        kind=trace.SpanKind.SERVER,
-        start_time=environ.get(_ENVIRON_STARTTIME_KEY),
-    )
-    if span.is_recording():
-        attributes = otel_wsgi.collect_request_attributes(environ)
-        if flask.request.url_rule:
-            # For 404 that result from no route found, etc, we
-            # don't have a url_rule.
-            attributes["http.route"] = flask.request.url_rule.rule
-        for key, value in attributes.items():
-            span.set_attribute(key, value)
-
-    activation = tracer.use_span(span, end_on_exit=True)
-    activation.__enter__()
-    environ[_ENVIRON_ACTIVATION_KEY] = activation
-    environ[_ENVIRON_SPAN_KEY] = span
-    environ[_ENVIRON_TOKEN] = token
+        environ = flask.request.environ
```

Comment:
I think we should pass in `environ` as an argument to the span callback, so users can change the span name based off these parameters.

Reply:
Yes, I thought about it. The request context `flask.request` is accessible for users just like how we are using it in instrumentation. Passing `environ` wouldn't really give any extra benefit. What do you think?

Reply:
Ahh that is true. Yeah maybe we don't need it in the params then.